File: highgo/src/test/java/com/dbx/agent/highgo/HighgoAgentTest.java
package com.dbx.agent.highgo;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class HighgoAgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new HighgoAgent();
    }

    @Override
    protected String resultSetSql() {
        return "CALL sample_proc()";
    }

    @Test
    void declaresHighgoPostgresLikeProfile() {
        HighgoAgent agent = new HighgoAgent();

        Assertions.assertEquals("com.highgo.jdbc.Driver", agent.getProfile().getDriverClass());
        Assertions.assertEquals("jdbc:highgo://{host}:{port}/{database}", agent.getProfile().getUrlTemplate());
    }
}

File: highgo/src/test/kotlin/com/dbx/agent/highgo/HighgoAgentTest.kt
package com.dbx.agent.highgo

import com.dbx.agent.DatabaseAgent
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest
import kotlin.test.Test
import kotlin.test.assertEquals

class HighgoAgentTest : JdbcFakeExecutionBehaviorTest() {
    override fun createAgent(): DatabaseAgent {
        return HighgoAgent()
    }

    override fun resultSetSql(): String = "CALL sample_proc()"

    @Test
    fun `declares highgo postgres-like profile`() {
        val agent = HighgoAgent()

        assertEquals("com.highgo.jdbc.Driver", agent.profile.driverClass)
        assertEquals("jdbc:highgo://{host}:{port}/{database}", agent.profile.urlTemplate)
    }
}
